packages/pdfshift/src/Adaptor.js

```diff
@@ -0,0 +1,201 @@
+ *     <p>Total Sales: $42</p>
+ *   </body>
+ * </html>
+ * `);
+ * @example <caption>Generate a PDF with options</caption>
+ * generatePDF(
+ * `<html>
+ *   <body>
+ *     <h1>Sales Report</h1>
+ *     <p>Date: 2025-02-01</p>
+ *     <p>Total Sales: $42</p>
+ *   </body>
+ * </html>
+ * `, {
+ *   sandbox: true});
+ * @example <caption>Generate a PDF from a url</caption>
+ * generatePDF('https://www.example.com/', {
+ *   sandbox: true,
+ *   encode: false,
+ * });
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function generatePDF(htmlTemplateString, options) {
+  return async state => {
+    const [resolvedhtmlTemplateString, resolvedOptions] = expandReferences(
+      state,
+      htmlTemplateString,
+      options
+    );
+    console.log('Preparing a PDF generation request to PDFShift');
+
+    const response = await util.request(
+      state.configuration,
+      'POST',
+      '/convert/pdf',
+      {
+        body: {
+          source: resolvedhtmlTemplateString,
+          encode: true,
+          ...resolvedOptions,
+        },
+        parseAs: 'text',
+      }
+    );
+
+    return util.prepareNextState(state, response);
+  };
+}
+
+/**
+ * PDF Body Options object. {@link https://docs.pdfshift.io/#convert|See PDFShift documentation} for more details.
+ * @typedef {Object} BodyOptions
+ * @property {string} source - The HTML string or url to convert to PDF.
+ * @property {boolean} sandbox - Generates PDF documents in dev mode that doesn't count in the credits.
+ * @property {boolean} encode - Return the generated PDF in Base64 encoded format, instead of raw. Defaults to `true`
+ */
+
+/**
+ * Make a POST request to PDFShift
+ * @example <caption> Convert a HTML string to PDF</caption>
+ * post(
+ *   '/convert/pdf',
+ *   {
+ *     source: `<html>
+ *     <body style="font-family: Arial, sans-serif; font-size: 14px;">
+ *       <h1>Sales Report</h1>
+ *       <p>Date: 2025-02-01</p>
+ *       <p>Total Sales: $42</p>
+ *     </body>
+ *   </html>`,
+ *     sandbox: true
+ *   },
+ *   {
+ *     parseAs: 'text',
+ *   }
+ * );
+ * @function
+ * @public
+ * @param {string} path - Path to resource
+ * @param {BodyOptions} body - Object which will be attached to the POST body
+ * @param {RequestOptions} options - Optional request options
+ * @returns {Operation}
+ * @state {HttpState}
+ */
+export function post(path, body, options) {
+  console.log('Preparing a POST request to PDFShift');
+  return request('POST', path, body, options);
+}
+
+/**
+ * Make a general HTTP request to PDFShift
+ * @example <caption>Make a request to convert a HTML string to PDF</caption>
+ * request(
+ *   'POST',
+ *   '/convert/pdf',
+ *   {
+ *     source: `<html>
+ *       <body style="font-family: Arial, sans-serif; font-size: 14px;">
+ *         <h1>Sales Report</h1>
+ *         <p>Date: 2025-02-01</p>
+ *         <p>Total Sales: $42</p>
+ *       </body>
+ *     </html>`,
+ *     sandbox: true
+ *   },
+ *   {
+ *     parseAs: 'text',
```

Comment:
In this adaptor we should probably default parseAs to text, no? Does it even have a JSON API?

This is the problem that @mtuchi raises really. By adding APIs which we don't need but seem harmless, we quickly get into questions like this. And I really don't want to be spending much time on it.

Reply:
We do have a get method, but I have made it a default in request and generatePdf